compiler/resolution/functionResolution.cpp

```diff
@@ -5885,6 +5887,27 @@ static void resolveInitVar(CallExpr* call) {
     gdbShouldBreakHere();
   }
 
+  if (call->isPrimitive(PRIM_INIT_VAR_SPLIT_INIT)) {
+    // If it has runtime type, turn this back into assignment
+    if (dst->type->symbol->hasFlag(FLAG_HAS_RUNTIME_TYPE)) {
+      call->primitive = primitives[PRIM_NOOP];
+      CallExpr* assign = new CallExpr("=", dst, src);
+      call->getStmtExpr()->insertAfter(assign);
+      resolveExpr(assign);
+      return;
+    }
+    // Also, check that the types are compatible for split initialization
+    // with type inference.
+    if (call->numActuals() < 3)
+      if (dst->type != dtUnknown)
+        if (dst->type != srcType)
+          USR_FATAL_CONT(call, "Split initialization uses multiple types; "
+                               "another initialization has type %s "
+                               "but this initialization has type %s",
+                               toString(dst->type),
+                               toString(srcType));
```

Comment:
Is it reasonable to provide the other initialization's line number? Ex.
```cpp
USR_FATAL_CONT(call, "Split initialization uses multiple types");
USR_PRINT(call, "this initialization has type %s" ...);
USR_PRINT(the-other-AST, "but this initialization has type %s" ...);
```

Reply:
I think it would be reasonable to do so, but creating such an error message is actually not easy given the AST structure here. In particular, `the-other-AST` is not easily available. So I would like to leave it alone for this PR (of course we can try to improve it in the future).